src/gtk/glcanvas.cpp

```diff
@@ -350,4 +354,15 @@ void wxGLCanvas::GTKInitImplicitContext()
 
 #endif // WXWIN_COMPATIBILITY_2_8
 
+void wxGLCanvas::EnsureAvailable()
+{
+#ifdef GDK_WINDOWING_X11
+    if ( !GDK_IS_X11_DISPLAY(gdk_display_get_default()) )
+#endif
+    {
+        wxString msg(wxT("wxGLCanvas is only supported on X11 currently.  You may be able to\nwork around this by setting environment variable GDK_BACKEND=x11 before starting\nyour program."));
```

Comment:
Why use this temporary string instead of passing it directly to `wxLogError()`?

Also, this should normally be translated, i.e. use `_(...)` around it.